tiledb/sm/rtree/rtree.cc

```diff
@@ -374,27 +381,36 @@ void RTree::deserialize_v5(Deserializer& deserializer, const Domain* domain) {
   auto level_num = deserializer.read<unsigned>();
 
   levels_.clear();
-  levels_.resize(level_num);
+  levels_.reserve(level_num);
   auto dim_num = domain->dim_num();
 
   for (unsigned l = 0; l < level_num; ++l) {
     auto mbr_num = deserializer.read<uint64_t>();
-    levels_[l].resize(mbr_num);
+    // We use emplace_back to construct the level in place, which also
+    // implicitly passes the memory tracking allocator all the way down to the
+    // NDRange objects.
+    auto& level = levels_.emplace_back();
+    level.reserve(mbr_num);
 
     for (uint64_t m = 0; m < mbr_num; ++m) {
-      levels_[l][m].resize(dim_num);
+      auto& mbr = level.emplace_back();
+      mbr.reserve(dim_num);
       for (unsigned d = 0; d < dim_num; ++d) {
         auto dim{domain->dimension_ptr(d)};
         if (!dim->var_size()) {  // Fixed-sized
           auto r_size = 2 * dim->coord_size();
           auto data = deserializer.get_ptr<void>(r_size);
-          levels_[l][m][d] = Range(data, r_size);
+          // NDRange is not yet a pmr vector, so we need to explicitly pass the
```

Comment:
Making `NDRange` a `tdb::pmr::pmr_vector` would require changing many files that use `std::vector<Range>` to use `NDRange`. I can change them, but am wary about expanding the PR's scope.

Reply:
Let's file a follow up story here.